test/TestGrains/ReentrantGrain.cs

```diff
@@ -68,6 +71,84 @@ public Task SetSelf(INonReentrantGrain self)
         }
     }
 
+    [MayInterleave("MayInterleave")]
```

Comment:
Does `[MayInterleave(nameof(MayInterleave)]` compile at this scope (or call the method MayInterleavePredicate to make it more explicit that it has hothing to do with the attribute name)? If so, that would be preferred, as people might look at this test to get a sense of how the feature works